pkg/asset/internal/templates.go

```diff
@@ -141,6 +141,7 @@ spec:
         - --tls-private-key-file=/etc/kubernetes/secrets/apiserver.key
         - --service-account-key-file=/etc/kubernetes/secrets/service-account.pub
         - --client-ca-file=/etc/kubernetes/secrets/ca.crt
+        - --cloud-provider={{ .CloudProvider }}
```

Comment:
What if `.CloudProvider == ""`? Perhaps these need to be wrapped in a conditional check?